duckduckgo/duckduckgo-privacy-extension
# Unhack the UI constants code

**Reviewer:** @dharb 
**Depends on:** #149 

## Description:

**NOTE** this doesn't merge back into develop.

This fixes the hacky UI code that I mentioned in [this PR](https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/149#discussion_r170289601).

Basically, even though the UI code uses Browserify under the hood, it accesses constants using `window.constants` for whatever reason. This PR makes sure it uses `require`.

This branch is branched off of #149 which is why the diff be crazy. See https://github.com/duckduckgo/duckduckgo-privacy-extension/compare/andrey/actually-browserify...andrey/ui-constants to see the diff before #149 gets merged.

## Steps to test this PR:

1. Check the truncated top trackers view shows all the major trackers correctly
2. Check TOSDR looks good everywhere when the site we're on has unknown privacy practices (on the main popup view, on the grade scorecard view and on the privacy practices view itself)
3. Check logic to do with major trackers is still solid (e.g. go to https://accounts.google.com and verify it's still showing as a D)
4. `window.constants` should not be a thing ever again

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
